controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -375,7 +381,14 @@ public String getScopeName() {
 
                         log.info("Scale {}/{} for segments started. Creating new segments. SegmentsToSeal {}", scope, name, sealedSegments);
                         // fresh run
-                        return scaleCreateNewSegments(newRanges, scaleTimestamp, segmentTable, activeEpoch);
+                        // Ensure that segment.creation time is monotonically increasing after each new scale
+                        long lastScaleTime = HistoryRecord.readLatestRecord(historyTable.getData(), true).map(HistoryRecord::getScaleTime).orElse(0L);
+
+                        long scaleEventTime = Math.max(System.currentTimeMillis(), scaleTimestamp);
```

Comment:
`scaleTimestamp` must have occurred in the past, so the max will always be `System.currentTimeMillis()`. `scaleTimestamp` will never be larger.

Reply:
Because of controller clock skews, a scaleTime set by another controller can be higher than the System.currentTimeInMillis on this controller.
Hence we take the max of prevScaleTS+1, System.currentTimeInMillis and scaleTimestamp

Reply:
ok, makes sense, but let's add a comment explaining what's going with this logic, please.